platform/Arg.roc

```diff
@@ -1,10 +1,42 @@
 module [
-    list!,
+    Arg,
+    display,
+    to_os_raw,
+    from_os_raw,
 ]
 
-import Host
+import InternalArg
 
-## Gives a list of the program's command-line arguments.
-list! : {} => List Str
-list! = \{} ->
-    Host.args! {}
+# os_str's are not necessarily valid utf-8 or utf-16
+# so we store as raw bytes internally to avoid
+# common mistakes
+Arg := InternalArg.ArgToAndFromHost
+
+## Unwrap the raw bytes for decoding, typically this is
+## consumed by a package and not an end user
+to_os_raw : Arg -> [Unix (List U8), Windows (List U16)]
+to_os_raw = \@Arg inner ->
+    when inner.type is
+        Unix -> Unix inner.unix
+        Windows -> Windows inner.windows
+
+from_os_raw : InternalArg.ArgToAndFromHost -> Arg
```

Comment:
Also, this should probably take `[Unix (List U8), Windows (List U16)]` to not expose internals.